### PR TITLE
[lg-1] Add file appender

### PR DESCRIPTION
**Pull request by OleksandrGubariev**
Configure file appender for logback.
1. All logs, starting from level DEBUG in your application, should be stored in file.
2. For file name, use next pattern:
- current log: photostudio.log
- previous logs: photostudio-[yyyy-MM-dd].log[.log_number], for example:
photostudio-2017-10-15.log.0 - for first log file which was published on 2017-10-15
3. Maximum size of log file is 5mb.

### CODE REVIEW

File: src/main/java/com/photostudio/dao/jdbc/JdbcOrderDao.java [hunk 1]
```diff
@@ -55,7 +61,7 @@ public JdbcOrderDao(DataSource dataSource) {
         if (resultWhere.contains("?")) {
 
             String selectOrdersByParameters = addSort(GET_ALL_ORDERS + resultWhere);
-
+            LOG.info("Get orders by parameters from DB");
```
Comment (Brutt) [inline src/main/java/com/photostudio/dao/jdbc/JdbcOrderDao.java]: Place this row at the very beginning of the method. Here add debug with incoming parameters.

File: src/main/java/com/photostudio/dao/jdbc/JdbcUserDao.java [hunk 2]
```diff
@@ -90,15 +98,17 @@ public void add(User user) {
...
+            LOG.debug("Add user: {} to DB", user);
         } catch (SQLException e) {
+            LOG.error("An exception occurred while trying to add user: {} DB", user, e);
             throw new RuntimeException("Can't add new user to DB ", e);
         }
     }
 
     @Override
     public User getUserById(long id) {
+        LOG.info("Get user by id");
```
Comment (Brutt) [inline src/main/java/com/photostudio/dao/jdbc/JdbcUserDao.java]: Add incoming param to log

File: src/main/java/com/photostudio/security/impl/DefaultSecurityService.java [hunk 3]
```diff
@@ -7,31 +7,39 @@
...
 
 public class DefaultSecurityService implements SecurityService {
+    private final Logger LOG = LoggerFactory.getLogger(getClass());
     private UserService userService = ServiceLocator.getService(UserService.class);
     private List<Session> sessionList = new CopyOnWriteArrayList<>();
 
     @Override
     public Session login(String login, String password) {
+        LOG.info("Connect user by login: {}", login);
+        LOG.debug("Connect user by login: {} {}", login, password);
```
Comment (Brutt) [inline src/main/java/com/photostudio/security/impl/DefaultSecurityService.java]: remove password from logs

File: src/main/java/com/photostudio/service/impl/DefaultUserService.java [hunk 4]
```diff
@@ -15,31 +18,38 @@ public DefaultUserService(UserDao userDao) {
 
     @Override
     public List<User> getAllUsers() {
-        return userDao.getAllUsers();
+        List<User> allUsers = userDao.getAllUsers();
+        LOG.info("Started service get all users from DB: {}", allUsers);
```
Comment (Brutt) [inline src/main/java/com/photostudio/service/impl/DefaultUserService.java]: change to Debug/Trace

File: src/main/java/com/photostudio/util/PropertyReader.java [hunk 5]
```diff
@@ -14,22 +18,25 @@ public PropertyReader(String path) {
 
     public Properties getProperties() {
         String prodEnvironment = System.getenv("environment");
-
+        LOG.info("System environment: {}", prodEnvironment);
```
Comment (Brutt) [inline src/main/java/com/photostudio/util/PropertyReader.java]: no need this log

File: src/main/java/com/photostudio/util/PropertyReader.java [hunk 6]
```diff
@@ -14,22 +18,25 @@ public PropertyReader(String path) {
...
 
     private Properties getProdProperties() {
         try {
             Properties properties = new Properties();
-
-            properties.setProperty("jdbc.url", System.getenv("JDBC_DATABASE_URL"));
-
+            String dbUrl = System.getenv("JDBC_DATABASE_URL");
+            properties.setProperty("jdbc.url", dbUrl);
+            LOG.info("Set jdbc.url: {}", dbUrl);
```
Comment (Brutt) [inline src/main/java/com/photostudio/util/PropertyReader.java]: change to debug

File: src/main/java/com/photostudio/util/PropertyReader.java [hunk 7]
```diff
@@ -40,9 +47,11 @@ private Properties getDevProperties() {
                 throw new IllegalArgumentException("No properties on path " + path);
             }
             properties.load(inputStream);
+            LOG.info("Read properties from path: {}", path);
```
Comment (Brutt) [inline src/main/java/com/photostudio/util/PropertyReader.java]: change to debug

File: src/main/java/com/photostudio/web/servlet/user/UserServlet.java [hunk 8]
```diff
@@ -43,6 +48,7 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 
     @Override
     protected void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        LOG.info("Request with user data for registration user received");
```
Comment (Brutt) [inline src/main/java/com/photostudio/web/servlet/user/UserServlet.java]: Think about to log all params with trace

File: src/main/java/com/photostudio/exception/LoginPasswordInvalidException.java [hunk 9]
```diff
@@ -1,7 +1,14 @@
...
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
 public class LoginPasswordInvalidException extends RuntimeException {
+    private final Logger LOG = LoggerFactory.getLogger(getClass());
+
     public LoginPasswordInvalidException(String message) {
+
         super(message);
+        LOG.info("Run LoginPasswordInvalidException with message: {}", message);
```
Comment (viktorterljuk) [inline src/main/java/com/photostudio/exception/LoginPasswordInvalidException.java]: I am not sure that logging inside an exception is a good idea. When it occurs you'll definitely know it without a log message.

Comment (Rooman) [reply src/main/java/com/photostudio/exception/LoginPasswordInvalidException.java]: This is hyevaya idea.

File: src/main/java/com/photostudio/security/impl/DefaultSecurityService.java (hunk 3 above)
Comment (viktorterljuk) [inline src/main/java/com/photostudio/security/impl/DefaultSecurityService.java]: Are you sure you need both of these? The better message, in this case, will be something like "Trying to log in the user with login: ... and password: ...".

File: src/main/java/com/photostudio/web/listener/ThymeleafInitializerListener.java [hunk 10]
```diff
@@ -1,15 +1,19 @@
...
 
 import javax.servlet.ServletContext;
 import javax.servlet.ServletContextEvent;
 import javax.servlet.ServletContextListener;
 
 public class ThymeleafInitializerListener implements ServletContextListener {
+    private final Logger LOG = LoggerFactory.getLogger(getClass());
     @Override
     public void contextInitialized(ServletContextEvent sce) {
         System.out.println("Initializing thymeleaf processor");
```
Comment (viktorterljuk) [inline src/main/java/com/photostudio/web/listener/ThymeleafInitializerListener.java]: As logging added, you don't need this line anymore.

File: src/main/java/com/photostudio/security/impl/DefaultSecurityService.java [hunk 11]
```diff
@@ -7,31 +7,39 @@
...
             if (!hashedPassword.equals(actualPassword)) {
-                throw new LoginPasswordInvalidException("Login/password invalid for user" + login);
+                LOG.error("Password invalid for user: {}", login);
+                throw new LoginPasswordInvalidException("Login/password invalid for user: " + login);
             }
             String userToken = UUID.randomUUID().toString();
             Session session = Session.builder().user(user)
                     .token(userToken).expireDate(LocalDateTime.now().plusHours(2)).build();
             sessionList.add(session);
+            LOG.info("User with login: {} is logged in ", login);
```
Comment (ValentynaBondarenko) [inline src/main/java/com/photostudio/security/impl/DefaultSecurityService.java]: should there be LOG.info or LOG.debug?

File: src/main/java/com/photostudio/dao/jdbc/JdbcUserDao.java [hunk 12]
```diff
@@ -68,14 +73,18 @@ public JdbcUserDao(DataSource dataSource) {
                 User user = USER_ROW_MAPPER.mapRow(resultSet);
                 users.add(user);
             }
+            LOG.info("Get: {} users from DB", users.size());
+            LOG.debug("Get users: {}", users);
             return users;
         } catch (SQLException e) {
-            throw new RuntimeException("Can't show all users", e);
+            LOG.error("An exception occurred while trying to get all users", e);
+            throw new RuntimeException("Can't show all users");
```
Comment (Brutt) [inline src/main/java/com/photostudio/dao/jdbc/JdbcUserDao.java]: backthrow new RuntimeException("Can't show all users"**, e**);

File: src/main/java/com/photostudio/dao/jdbc/JdbcUserDao.java [hunk 13]
```diff
@@ -110,10 +122,13 @@ public User getUserById(long id) {
...
                     throw new RuntimeException("More than one users found");
                 }
+                LOG.info("Getting user by id: {} is completed", id);
+                LOG.debug("Get user: {} by id: {}", user, id);
                 return user;
             }
         } catch (SQLException e) {
-            throw new RuntimeException("Can't found user where id=" + id, e);
+            LOG.error("An exception occurred while trying to get user by id: {}", id, e);
+            throw new RuntimeException("Can't found user where id=" + id);
```
Comment (Brutt) [inline src/main/java/com/photostudio/dao/jdbc/JdbcUserDao.java]: throw new RuntimeException("Can't found user where id=" + id, **e**);

File: src/main/java/com/photostudio/dao/jdbc/JdbcUserDao.java [hunk 14]
```diff
@@ -123,17 +138,21 @@ public void edit(User user) {
...
+        LOG.info("Delete user by id: {}", id);
         try (Connection connection = dataSource.getConnection();
              PreparedStatement preparedStatement = connection.prepareStatement(DELETE_USER)) {
             preparedStatement.setLong(1, id);
             preparedStatement.executeUpdate();
+            LOG.info("Deleting user by id: {} is completed", id);
         } catch (SQLException e) {
-            throw new RuntimeException("Can't remove user", e);
+            LOG.error("An exception occurred while trying to delete user by id: {}", id, e);
+            throw new RuntimeException("Can't remove user");
```
Comment (Brutt) [inline src/main/java/com/photostudio/dao/jdbc/JdbcUserDao.java]: throw new RuntimeException("Can't remove user", **e**);

File: src/main/java/com/photostudio/dao/jdbc/JdbcUserDao.java [hunk 15]
```diff
@@ -145,16 +164,21 @@ public User getByLogin(String login) {
...
                     throw new RuntimeException("More then one user found");
                 }
+                LOG.info("Getting user by login: {} is completed", login);
+                LOG.debug("Get user: {} by login: {}",user, login);
                 return user;
             }
         } catch (SQLException e) {
-            throw new RuntimeException("Get user by login error", e);
+            LOG.error("An exception occurred while trying to get user with login: {}", login, e);
+            throw new RuntimeException("Get user by login error");
```
Comment (Brutt) [inline src/main/java/com/photostudio/dao/jdbc/JdbcUserDao.java]: throw new RuntimeException("Get user by login error", **e**);

File: src/main/java/com/photostudio/exception/LoginPasswordInvalidException.java [hunk 16]
```diff
@@ -1,6 +1,11 @@
 package com.photostudio.exception;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
 public class LoginPasswordInvalidException extends RuntimeException {
+    private final Logger LOG = LoggerFactory.getLogger(getClass());
```
Comment (Brutt) [inline src/main/java/com/photostudio/exception/LoginPasswordInvalidException.java]: no need it

File: src/main/java/com/photostudio/security/impl/DefaultSecurityService.java [hunk 17]
```diff
@@ -7,31 +7,38 @@
...
     @Override
     public Session login(String login, String password) {
+        LOG.info("Connect user by login: {}", login);
         User user = userService.getUserByLogin(login);
         if (user != null) {
             String hashedPassword = getHashedPassword(user.getSalt(), password);
             String actualPassword = user.getPasswordHash();
             if (!hashedPassword.equals(actualPassword)) {
-                throw new LoginPasswordInvalidException("Login/password invalid for user" + login);
+                LOG.error("Password invalid for user: {}", login);
```
Comment (Brutt) [inline src/main/java/com/photostudio/security/impl/DefaultSecurityService.java]: This message reveals the true cause that **password** is invalid. Change to "Login/password invalid for user"

File: src/main/java/com/photostudio/service/impl/DefaultUserService.java [hunk 18]
```diff
@@ -15,31 +18,38 @@ public DefaultUserService(UserDao userDao) {
 
     @Override
     public List<User> getAllUsers() {
-        return userDao.getAllUsers();
+        List<User> allUsers = userDao.getAllUsers();
+        LOG.debug("Started service get all users from DB: {}", allUsers);
```
Comment (Brutt) [inline src/main/java/com/photostudio/service/impl/DefaultUserService.java]: Why do you use info in DefaultOrderService and here use debug?

File: src/main/java/com/photostudio/util/PropertyReader.java [hunk 19]
```diff
@@ -40,9 +47,11 @@ private Properties getDevProperties() {
                 throw new IllegalArgumentException("No properties on path " + path);
             }
             properties.load(inputStream);
+            LOG.debug("Read properties from path: {}", path);
             return properties;
         } catch (IOException e) {
-            throw new RuntimeException("Can't read properties file " + path, e);
+            LOG.error("Can't read properties file: {} ", path, e);
+            throw new RuntimeException("Can't read properties file " + path);
```
Comment (Brutt) [inline src/main/java/com/photostudio/util/PropertyReader.java]: throw new RuntimeException("Can't read properties file " + path, **e**);

File: src/main/java/com/photostudio/web/servlet/AdminPageServlet.java [hunk 20]
```diff
@@ -11,18 +13,20 @@
...
         try {
             Map<String, Object> paramsMap = new HashMap<>();
             commonVariableAppendService.appendUser(paramsMap, request);
             response.setContentType("text/html;charset=utf-8");
-
             TemplateEngineFactory.process("admin-page", paramsMap, response.getWriter());
         } catch (IOException e) {
-            throw new RuntimeException("AdminPageServlet error", e);
+            LOG.error("AdminPageServlet error", e);
+            throw new RuntimeException("AdminPageServlet error");
```
Comment (Brutt) [inline src/main/java/com/photostudio/web/servlet/AdminPageServlet.java]: throw new RuntimeException("AdminPageServlet error", **e**);

File: src/main/java/com/photostudio/web/servlet/LoginServlet.java [hunk 21]
```diff
@@ -50,7 +55,8 @@ protected void doPost(HttpServletRequest request, HttpServletResponse response)
                 TemplateEngineFactory.process("login", paramsMap, response.getWriter());
             }
         } catch (IOException e) {
-            throw new RuntimeException("LoginServlet doPost error", e);
+            LOG.error("LoginServlet doPost error", e);
+            throw new RuntimeException("LoginServlet doPost error");
```
Comment (Brutt) [inline src/main/java/com/photostudio/web/servlet/LoginServlet.java]: throw new RuntimeException("LoginServlet doPost error", **e**);

File: src/main/java/com/photostudio/web/servlet/order/AllOrdersServlet.java [hunk 22]
```diff
@@ -32,7 +35,8 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response) {
 
             TemplateEngineFactory.process("all-orders", paramsMap, response.getWriter());
         } catch (IOException e) {
-            throw new RuntimeException("AllOrdersServlet error", e);
+            LOG.error("AllOrdersServlet error", e);
+            throw new RuntimeException("AllOrdersServlet error");
```
Comment (Brutt) [inline src/main/java/com/photostudio/web/servlet/order/AllOrdersServlet.java]: throw new RuntimeException("AllOrdersServlet error", **e**);

**Review by Yuliia-Sh — DISMISSED** [review]

**Review by Brutt — APPROVED** [review]

**Review by ValentynaBondarenko — APPROVED** [review]

**Review by Yuliia-Sh — APPROVED** [review]

**Pull request merged**